### PR TITLE
Add quit function

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

**Review by godsflaw — APPROVED** [review]

**Comment by livnev** [comment]
Worth noting that there's an alternative, which is to send it to the surplus buffer in the Vow. However, if that doesn't happen before the `thaw` period is over, then it will be lost (instead of getting burned and affecting the global settlement calculation to slightly increase amount of every `gem` claimable). Therefore, this seems like the reasonable thing to do in this case.

File: src/RwaUrn.sol [hunk 1]
```diff
@@ -149,4 +150,14 @@ contract RwaUrn {
         vat.frob(ilk, address(this), address(this), address(this), 0, -int(dart));
         emit Wipe(wad);
     }
+
+    // If Dai is sitting here after ES that should be sent back
+    function quit() external {
+        require(outputConduit != address(0));
+        require(vat.live() == 0, "RwaUrn/overflow");
```
Comment (livnev) [inline src/RwaUrn.sol]: Fix revert string here

**Comment by livnev** [comment]
Also worth remarking that `RwaUrn.free()` is still expected to work after `cage`.

**Review by livnev — CHANGES_REQUESTED** [review]

**Comment by gbalabasquer** [comment]
Yes exactly, we could join the Dai and send it to the `vow` if `end.debt == 0` but not sure if worth it to complicate it more to win that window of time.

File: src/RwaUrn.sol (hunk 1 above)
Comment (gbalabasquer) [reply src/RwaUrn.sol]: Fixed

**Review by livnev — APPROVED** [review]
LGTM.

**Pull request merged**